---
title: "osca4anvil: enabling an AnVIL workspace for Orchestrating Single Cell Analysis with Bioconductor"
author: "Vincent J. Carey, stvjc at channing.harvard.edu"
date: "`r format(Sys.time(), '%B %d, %Y')`"
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{osca4anvil: enabling an AnVIL workspace for Orchestrating Single Cell Analysis with Bioconductor}
  %\VignetteEncoding{UTF-8}
output:
  BiocStyle::html_document:
    highlight: pygments
    number_sections: yes
    theme: united
    toc: yes
---

# Introduction

This package includes scripts that acquire and install selected versions
of the OSCA book for single-cell genomics.


## Approach using a default cloud environment in AnVIL


The following steps work with sources from Bioconductor 3.19.
```
BiocManager::install("vjcitn/osca4anvil", ref="bioc319")
library(osca4anvil)
clone_osca(branch="RELEASE_3_19")
lapply(osca4anvil::packs(), osca4anvil::get_deps, 
  installer=function(x) BiocManager::install(x, ask=FALSE))
```

Limited confirmation:
```
library(scRNAseq)
sce.416b <- LunSpikeInData(which="416b")
library(AnnotationHub)
ens.mm.v97 <- AnnotationHub()[["AH73905"]]
chr.loc <- mapIds(ens.mm.v97, keys=rownames(sce.416b),
keytype="GENEID", column="SEQNAME")
is.mito <- which(chr.loc=="MT")
library(scuttle)
df <- perCellQCMetrics(sce.416b, subsets=list(Mito=is.mito))
low.lib <- isOutlier(df$sum, type="lower", log=TRUE)
table(low.lib)
```
